### ISSUE TITLE
Support property-level extending for array properties

### ISSUE THREAD

**Issue by Majkl578** [squizlabs/PHP_CodeSniffer#2153]
Adding new property entries for array properties (i.e. ForbiddenFunctions and more in user-land) is cumbersome as one has to copy&paste all previous values and add new.

My proposal is to allow extending previously declared property values by adding new entries instead of replacing them:
base ruleset:
```xml
<rule ref="Generic.PHP.ForbiddenFunctions">
    <properties>
        <property name="forbiddenFunctionNames" type="array">
            <element key="foo" value="fixedFoo"/>
        </property>
    </properties>
</rule>
```

extending (project-level) ruleset:
```xml
<rule ref="Generic.PHP.ForbiddenFunctions">
    <properties>
        <property name="forbiddenFunctionNames" type="array" extend="true">
            <element key="bar" value="fixedBar"/>
        </property>
    </properties>
</rule>
```

This would result in Generic.PHP.ForbiddenFunctions->forbiddenFunctionNames having two entries:
```
foo => fixedFoo
bar => fixedBar
```

This would also overwrite previously declared `foo` if extending ruleset defined it. For lists, it would just append the new entries.

**Comment by gsherwood** [comment]
I think this is a good idea and the implementation seems fine to me. I'll put this on the roadmap. Thanks.

**Comment by gsherwood** [comment]
Added as part of PR #2154
